src/OrchardCore.Modules/OrchardCore.Email/Controllers/AdminController.cs

```diff
@@ -1,31 +1,36 @@
-using System;
+using System.Collections.Generic;
+using System.Linq;
 using System.Threading.Tasks;
 using Microsoft.AspNetCore.Authorization;
 using Microsoft.AspNetCore.Mvc;
 using Microsoft.AspNetCore.Mvc.Localization;
 using OrchardCore.DisplayManagement.Notify;
 using OrchardCore.Email.Drivers;
+using OrchardCore.Email.Services;
 using OrchardCore.Email.ViewModels;
+using OrchardCore.Modules;
 
 namespace OrchardCore.Email.Controllers
 {
+    [Feature("OrchardCore.Email.Smtp")]
     public class AdminController : Controller
     {
         private readonly IAuthorizationService _authorizationService;
         private readonly INotifier _notifier;
-        private readonly ISmtpService _smtpService;
+        private readonly IEmailService _emailService;
         protected readonly IHtmlLocalizer H;
 
         public AdminController(
             IHtmlLocalizer<AdminController> h,
             IAuthorizationService authorizationService,
             INotifier notifier,
-            ISmtpService smtpService)
+            IEnumerable<IEmailService> emailServices)
         {
             H = h;
             _authorizationService = authorizationService;
             _notifier = notifier;
-            _smtpService = smtpService;
+            // Gets SmtpEmailService instance in case there're many IEmailService implementations registered in the DI container
+            _emailService = emailServices.OfType<SmtpEmailService>().Single();
```

Comment:
FYI this could coz an exception if the `AzureEmailService` registered later. For that, I get all the registered implementations and then filter them to avoid testing against another `IEmailService` implementation.

Reply:
You're over complicating the implementation. You should follow my previous suggestion so you don't have this logic. You can use a service factory/resolver that would return the correct instances based on the app settings.

Reply:
I will check your comments tomorrow

Reply:
I used the KeyedService, which was suited in this case